PlayniteSounds.cs

```diff
@@ -169,878 +284,380 @@ private void onWindowStateChanged(object sender, EventArgs e)
+            try
+            {
+                InitialCopyAudioFiles();
+
+                if (_closeAudiofilesNextPlay)
+                {
+                    CloseAudioFiles();
+                    _closeAudiofilesNextPlay = false;
+                }
+
+                PlayerEntry Entry;
+                if (_players.ContainsKey(FileName))
+                {
+                    Entry = _players[FileName];
+                }
+                else
+                {
+                    string FullFileName = Path.Combine(GetPluginUserDataPath(), SoundDirectory.Sound, FileName);
+
+                    //MediaPlayer can play multiple sounds together from mulitple instances SoundPlayer can not
+                    var fileExists = File.Exists(FullFileName);
+                    Entry = UseSoundPlayer
+                        ? new PlayerEntry(fileExists, new SoundPlayer())
+                        : new PlayerEntry(fileExists, new MediaPlayer());
+
+                    if (fileExists)
                     {
-                        OpenSoundsFolder();
+                        if (Entry.IsMediaPlayer)
+                        {
+                            Entry.MediaPlayer.Open(new Uri(FullFileName));
+                        }
+                        else
+                        {
+                            Entry.SoundPlayer.SoundLocation = FullFileName;
+                            Entry.SoundPlayer.Load();
+                        }
                     }
-                },
-                new MainMenuItem {
-                    MenuSection = "@Playnite Sounds",
-                    Icon = Path.Combine(pluginFolder, "icon.png"),
-                    Description = resources.GetString("LOC_PLAYNITESOUNDS_ActionsReloadAudioFiles"),
-                    Action = (MainMenuItem) =>
+                    _players[FileName] = Entry;
+                }
+
+                if (Entry.FileExists)
+                {
+                    if (Entry.IsMediaPlayer)
                     {
-                        ReloadAudioFiles();
+                        Entry.MediaPlayer.Stop();
+                        Entry.MediaPlayer.Play();
                     }
-                },
-                new MainMenuItem {
-                    MenuSection = "@Playnite Sounds",
-                    Icon = Path.Combine(pluginFolder, "icon.png"),
-                    Description = resources.GetString("LOC_PLAYNITESOUNDS_ActionsHelp"),
-                    Action = (MainMenuItem) =>
+                    else
                     {
-                        HelpMenu();
+                        Entry.SoundPlayer.Stop();
+                        Entry.SoundPlayer.PlaySync();
                     }
                 }
-             };
-            return MainMenuItems;
+            }
+            catch (Exception E)
+            {
+                HandleException(E);
+            }
         }
 
-        public void HelpMenu()
+        private bool ShouldPlayMusic()
```

Comment:
Not sure how I feel about ShouldPlayMusic looking like a simple check, but having the CloseMusic side effect. Might be too sneaky.